keps/sig-architecture/4330-compatibility-versions/README.md

```diff
@@ -507,6 +508,27 @@ compatibility version can be set to.
 
 Alpha APIs may not be enabled in conjunction with compatibility version.
 
+### Discovery
+
+Discovery will [enable](https://github.com/kubernetes/kubernetes/blob/7080b51ee92f67623757534f3462d8ae862ef6fe/staging/src/k8s.io/apiserver/pkg/util/openapi/enablement.go#L32) the group versions matching the compatibility version.
+
+API fields that were introduced after the compatibility version will **not** be
+pruned. There is a tradoff here:
```

Comment:
AFAIK discovery doesn't expose any api fields. Is this referring to the OpenAPI schemas?

Reply:
Yes, apologies if my terminology is off.. I think of discovery including both api discovery and openapi schemas.. Happy to take suggestions on the correct use of terms.

Reply:
Gotcha, since they are separate documents with different sources of truth and usages I think it would makes sense to separate them in the KEP. 

IMO my gut instinct is that it should be fine to leave inert fields exposed in the schema, unless they are listed as "required". Though I think in general we do not make a habit out of adding new `required` fields.

Reply:
>  not make a habit out of adding new required fields. It would be nice if there was a way to mark the field as ignored

Yep, we can design with the expectation that all of https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md is respected. In particular these rules:

1. Any API call (e.g. a structure POSTed to a REST endpoint) that succeeded before your change must succeed after your change.
2. Any API call that does not use your change must behave the same as it did before your change.

Reply:
Offline it sounded like we have the ability to enumerate these fields that are exposed-but-ignored due to their features being disabled by the compatibiltiy version flag. Not saying we should, but if we wanted to, would we be able to mark such fields in the schema in an x-extension?

Reply:
I like the idea. It would be non-trivial sing we don't track feature gated fields in a declarative way. Maybe handle as a separate enhancement?